src/kafka/producer.js

```diff
@@ -283,31 +352,48 @@ class Producer extends EventEmitter {
    * @returns {boolean} or if failed {Error}
    */
   async sendMessage (messageProtocol, topicConf) {
-    const { logger } = this._config
-    try {
-      if (!this._producer) {
-        throw new Error('You must call and await .connect() before trying to produce messages.')
-      }
-      if (this._producer._isConnecting) {
-        Logger.isDebugEnabled && logger.debug('still connecting')
-      }
-      const parsedMessage = Protocol.parseMessage(messageProtocol)
-      const parsedMessageBuffer = this._createBuffer(parsedMessage, this._config.options.messageCharset)
-      if (!parsedMessageBuffer || !(typeof parsedMessageBuffer === 'string' || Buffer.isBuffer(parsedMessageBuffer))) {
-        throw new Error('message must be a string or an instance of Buffer.')
+    return new Promise((resolve, reject) => {
+      const { logger } = this._config
+      try {
+        if (!this._producer) {
+          throw new Error('You must call and await .connect() before trying to produce messages.')
+        }
+        if (this._producer._isConnecting) {
+          Logger.isDebugEnabled && logger.debug('still connecting')
+        }
+        const parsedMessage = Protocol.parseMessage(messageProtocol)
+        const parsedMessageBuffer = this._config.options.serializeFn(parsedMessage, this._config.options) // this._createBuffer(parsedMessage, this._config.options.messageCharset)
+        if (!parsedMessageBuffer || !(typeof parsedMessageBuffer === 'string' || Buffer.isBuffer(parsedMessageBuffer))) {
+          throw new Error('message must be a string or an instance of Buffer.')
+        }
+        Logger.isDebugEnabled && logger.debug('Producer::send() - start %s', JSON.stringify({
```

Comment:
This type of interpolation won't work due to https://github.com/mojaloop/project/issues/3478
You'll have to use `${}`